client/src/components/Draftail/decorators/Link.js

```diff
@@ -47,6 +48,75 @@ export const getLinkAttributes = (data) => {
   };
 };
 
+/**
+ * See https://docs.djangoproject.com/en/4.0/_modules/django/core/validators/#EmailValidator.
+ */
+const djangoUserRegex =
+  /(^[-!#$%&'*+/=?^_`{}|~0-9A-Z]+(\.[-!#$%&'*+/=?^_`{}|~0-9A-Z]+)*$|^"([\001-\010\013\014\016-\037!#-[\]-\177]|\\[\001-\011\013\014\016-\177])*"$)/i;
+// Compared to Django, changed to remove the end-of-domain `-` check that was done with a negative lookbehind `(?<!-)` (unsupported in Safari), and disallow all TLD hyphens instead.
+// const djangoDomainRegex = /((?:[A-Z0-9](?:[A-Z0-9-]{0,61}[A-Z0-9])?\.)+)(?:[A-Z0-9-]{2,63}(?<!-))$/i;
+const djangoDomainRegex =
+  /((?:[A-Z0-9](?:[A-Z0-9-]{0,61}[A-Z0-9])?\.)+)(?:[A-Z0-9]{2,63})$/i;
+/**
+ * See https://docs.djangoproject.com/en/4.0/_modules/django/core/validators/#URLValidator.
+ */
+const djangoSchemes = ['http:', 'https:', 'ftp:', 'ftps:'];
+
+export const getValidLinkURL = (text, schemes) => {
```

Comment:
This is a fair bit different to Django’s URLValidator & EmailValidator, but hopefully only in ways that make this more restrictive.